lib/core/src/Cardano/Wallet.hs

```diff
@@ -1954,10 +1956,9 @@ joinStakePool
     -> PoolId
     -> PoolLifeCycleStatus
     -> WalletId
-    -> ArgGenChange s
     -> Passphrase "raw"
-    -> ExceptT ErrJoinStakePool IO (Tx, TxMeta, UTCTime)
-joinStakePool ctx currentEpoch knownPools pid poolStatus wid argGenChange pwd =
+    -> ExceptT ErrJoinStakePool IO (CoinSelection, [Cardano.Certificate], DelegationAction)
```

Comment:
In principle, we don't want to expose a type from `cardano-ledger-specs` or `cardano-api` in here. The code in Cardano.Wallet is supposed to be "backend-agnostic" and this `Cardano.Certificate` is specific to cardano-node (makes no sense for Jormungandr). 

Plus, isn't that redundant with the `DelegationAction` already ?

Reply:
How so? We want to serialise the certificates. So we have to know what types they are: https://input-output-hk.github.io/ouroboros-network/shelley-spec-ledger/Shelley-Spec-Ledger-Delegation-Certificates.html#t:DelegCert

Reply:
In addition, this endpoint won't be available for jormungandr anyway. How would we abstract over this?

Reply:
But the `DelegationAction` are the certificates already. They capture the exact same concept.

Reply:
So do you mean we keep:

```hs
    _mkDelegationCertificates
        :: DelegationAction
            -- Pool Id to which we're planning to delegate
        -> XPub
            -- Reward account public key
        -> [Cardano.Certificate]
    _mkDelegationCertificates da accXPub =
        case da of
           Join poolId ->
                   [ toStakePoolDlgCert accXPub poolId ]
           RegisterKeyAndJoin poolId ->
                   [ toStakeKeyRegCert  accXPub
                   , toStakePoolDlgCert accXPub poolId
                   ]
           Quit -> [toStakeKeyDeregCert accXPub]
```

private and not in TransactionLayer and then use it when we serialize DelegationAction in the shelley API?

Reply:
Yes. This is a shelley-specific detail so it shouldn't escape the shelley package.

Reply:
:godmode: